### PR TITLE
Leverage gradle toolchain only for gradle 6.7+

### PR DESCRIPTION
**Pull request by glefloch**
In #16351, I added support for gradle build tool chain. (https://docs.gradle.org/current/userguide/toolchains.html)

This is only available from gradle 6.7. Using a previous gradle version ends up with a `NoClassDefFoundError`. 
This branch check the gradle version in order to only leverage build tool chain for compatible gradle version.

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java [hunk 1]
```diff
@@ -210,11 +205,15 @@ private QuarkusDevModeLauncher newLauncher() throws Exception {
         final Project project = getProject();
 
         String java = null;
-        Provider<JavaLauncher> javaLauncher = getJavaLauncher();
-        if (javaLauncher.isPresent()) {
-            java = javaLauncher.get().getExecutablePath().getAsFile().getAbsolutePath();
-        }
 
+        if (toolchainSupported) {
```
Comment (gastaldi) [inline devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java:209]: Why another flag? Isn't easier to just change the condition here?

```suggestion
        if ((GradleVersion.current().compareTo(GradleVersion.version("6.7"))) {
```

Comment (glefloch) [reply devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java]: right, it's simpler :)

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by gsmet** [comment]
The other patch hasn't been backported so we either backport both or neither. @glefloch WDYT?

**Comment by glefloch** [comment]
I'm not sure we should back port those two, this can wait the next release.

**Pull request merged**